jans-cedarling/cedarling/src/jwt/error.rs

```diff
@@ -5,28 +5,70 @@
  * Copyright (c) 2024, Gluu, Inc.
  */
 
+use super::decoding_strategy::key_service;
+
 /// Error type for the JWT service.
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    /// Failed to parse the JWT.
-    #[error("error parsing the JWT: {0}")]
-    Parsing(#[source] jsonwebtoken::errors::Error),
-
-    /// Key could not be retrieved from the `KeyService`.
-    #[error("could not get hold of a key from the KeyService: {0}")]
-    MissingKey(String),
+    /// Indicates an error encountered while parsing the JWT.
+    ///
+    /// This variant occurs when the provided JWT cannot be properly parsed.
+    #[error("Error parsing the JWT: {0}")]
+    ParsingError(#[source] jsonwebtoken::errors::Error),
 
-    /// Tried to validate a token signed with an unsupported algorithm.
-    #[error("the JWT is signed with an unsupported algorithm: {0:?}")]
+    /// Indicates that the token was signed with an unsupported algorithm.
+    ///
+    /// This occurs when the JWT header specifies an algorithm that is not supported
+    /// by the current validation configuration.
+    #[error("The JWT is signed with an unsupported algorithm: {0:?}")]
     TokenSignedWithUnsupportedAlgorithm(jsonwebtoken::Algorithm),
 
-    /// The JWT is missing a `kid` header.
-    #[error("the JWT is missing a required header: {0}")]
+    /// Indicates that a required header is missing from the JWT.
+    ///
+    /// The JWT specification requires certain headers (e.g., `kid` for key identification).
+    /// This error is returned if any of those required headers are missing.
+    #[error("The JWT is missing a required header: {0}")]
     MissingRequiredHeader(String),
 
-    /// Failed to validate the JWT
-    #[error("failed to validate JWT: {0}")]
-    Validation(#[source] jsonwebtoken::errors::Error),
-}
+    /// Indicates failure during the validation of the JWT.
+    ///
+    /// This occurs when JWT validation fails due to issues such as an invalid signature,
+    /// claim validation failure, or other validation errors.
+    #[error("Failed to validate the JWT: {0}")]
+    ValidationError(String),
+
+    /// Indicates an unsupported algorithm defined in the configuration.
+    ///
+    /// This variant indicates that the configuration specifies an algorithm that
+    /// is not yet implemented or recognized by the JWT service.
+    #[error("An algorithm defined in the configuration is not yet implemented: {0}")]
+    UnimplementedAlgorithm(Box<str>),
```

Comment:
it happen only in `string_to_alg` and we no need to return this error when parse JWT, because it will newer happen.

Reply:
i'll fix it in #9945

Reply:
this issue got introduced because of a merge that removed the function using returning that error.

Reply:
actually, i changed my mind and just tried to manually revert the changes here: [c8c8945](https://github.com/JanssenProject/jans/pull/9939/commits/c8c8945883b97b1041fccbba0c10f3c9b17c4d19). should be back to it's original implementation.